Refactor locale selection dialog

lukstbit opened the pull request:
## Purpose / Description

Done mostly to remove the listener code form ModelFieldEditor but it also improves the ui and fixes some ux issues(like touch target size and touch feedback). I'll squash, made two commits to separate the(more important) fragment result changes from minor ui changes.

Before/After

<img width="320" height="750" src="https://github.com/user-attachments/assets/621767cf-1d61-48ad-94ed-17647f323982" /><img width="320" height="750" src="https://github.com/user-attachments/assets/5655f4ba-9858-4172-b240-fa60aa437eba"/>


## How Has This Been Tested?

Ran the tests, manually verified the dialog.

## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

david-allison commented:
Padding is off in the TextView still, should be consistent with the rest of the list

david-allison commented on AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.kt:
```diff
@@ -86,6 +85,19 @@ class ModelFieldEditor :
         startLoadingCollection()
+        supportFragmentManager.setFragmentResultListener(REQUEST_HINT_LOCALE_SELECTION, this) { _, bundle ->
+            // if the user cancelled the selection the Bundle will be empty
+            if (bundle.containsKey(KEY_SELECTED_LOCALE)) {
```
nit: I'd be tempted to invert for readability

david-allison reviewed (commented):
Looks great, cheers. One request on the padding

BrayanDSO reviewed (approved):
Incremental change. LGTM

david-allison reviewed (dismissed):
Changing my review above to a very mild 'requesting changes' to update the GitHub UI. 

Merge at will, once padding is fixed.

If there's any pushback on this request, merge at will

lukstbit commented:
I think this is ready now. Changes in the last push:

- fixed padding
- used the fragment result utility that I recently added
- fixed a typo(added to dictionary) issue in LocaleSelectionDialog to make it clean
- fixed David's review comment above by passing a null locale

<img width="320" height="720" src="https://github.com/user-attachments/assets/8c4681e7-ee10-4626-ba10-bb1875a96eb0"/>
